File: python-project-euler-100/p010.py
from Solution import Solution

class P010(Solution):
    def sieve_of_eratosthenes(self, n): 
          
        # Create a boolean array "prime[0..n]" and initialize 
        # all entries in it as true. A value in prime[i] will 
        # finally be false if i is Not a prime, else true. 
        prime = [True for i in range(n + 1)] 
        p = 2
        while (p * p <= n): 
              
            # If prime[p] is not changed, then it is a prime 
            if (prime[p] == True): 
                # Update all multiples of p 
                for i in range(p * 2, n + 1, p): 
                    prime[i] = False
            p += 1
        prime[0]= False
        prime[1]= False
        return [index for index,value in enumerate(prime) if prime[index]]
    
    def solve(self):
        self.problem_number = 10
        last = 2000000
        list_of_primes = self.sieve_of_eratosthenes(last)
        sum_of_primes = sum(list_of_primes)
    
        return sum_of_primes

def main():
	P010().run()

if __name__ == "__main__":
	main()

File: go-project-euler-100/p010.go
package main

import "math"

func sieveOfEratosthenes(n int) []int {
	expectedCapacity := int(math.Floor(math.Sqrt(float64(n))))
	notPrime := make([]bool, 0, n)
	for i := 0; i < n; i++ {
		notPrime = append(notPrime, false)
	}

	for p := 2; p*p <= n; p++ {

		// If notPrime[p] is not changed, then it is a notPrime
		if !notPrime[p] {
			// Update all multiples of p
			for i := (p * 2); i < n; i += p {
				notPrime[i] = true
			}
		}
	}
	notPrime[0] = true
	notPrime[1] = true

	listOfPrimes := make([]int, 0, expectedCapacity)
	for i := 0; i < n; i++ {
		if !notPrime[i] {
			listOfPrimes = append(listOfPrimes, i)
		}
	}

	return listOfPrimes
}

/*P010Solve solution for p010*/
func P010Solve() int {
	last := 2000000
	sum := 0
	listOfPrimes := sieveOfEratosthenes(last)
	for _, i := range listOfPrimes {
		sum += i
	}

	return sum
}
